malaria24/ona/tests/test_tasks.py

```diff
@@ -180,3 +185,93 @@ def test_ona_fetch_forms(self):
         self.assertEqual(form.title, 'the-form-title')
         self.assertEqual(form.form_id, '12345')
         self.assertEqual(form.active, False)
+
+    @responses.activate
+    def test_sms_vumi_go_if_channel_empty(self):
+        """
+        Checks vumi go is the default channel
+        """
+        send_sms(to='+27111111111', content='test message')
+
+        self.assertEqual(len(responses.calls), 1)
+        self.assertEqual(responses.calls[0].request.url,
+                         "http://go.vumi.org/api/v1/go/http_api_nostream"
+                         "/VUMI_GO_CONVERSATION_KEY/messages.json")
+        [sms] = SMS.objects.all()
+        self.assertEqual(sms.content, "test message")
+
+    @responses.activate
+    @override_settings(
+        SMS_CHANNEL='VUMI_GO')
+    def test_sms_vumi_go_if_set(self):
+        """
+        Checks sms sent via vumi go if that's what the setting specifies
+        """
+        send_sms(to='+27111111111', content='test message')
+
+        self.assertEqual(len(responses.calls), 1)
+        self.assertEqual(responses.calls[0].request.url,
+                         "http://go.vumi.org/api/v1/go/http_api_nostream"
+                         "/VUMI_GO_CONVERSATION_KEY/messages.json")
+        [sms] = SMS.objects.all()
+        self.assertEqual(sms.content, "test message")
+
+    @responses.activate
+    @override_settings(
+        SMS_CHANNEL='JUNEBUG')
+    def test_sms_junebug_fails_if_settings_missing(self):
+        """
+        Checks sending via Junebug requires extra settings
+        """
+        with self.assertRaises(AttributeError):
+            send_sms(to='+27111111111', content='test message')
+
+        with self.settings(JUNEBUG_URL='http://junebug.qa.malariaconnect.org'):
+            with self.assertRaises(AttributeError):
+                send_sms(to='+27111111111', content='test message')
+
+    @responses.activate
+    @override_settings(
+        SMS_CHANNEL='JUNEBUG',
+        JUNEBUG_CHANNEL_URL='http://jb.qa.malariaconnect.org/CHANNEL_ID')
```

Comment:
wouldn't it be safer to use a fake domain in tests? Just so that you're absolutely sure it will never reach its destination

Reply:
This is a fake one. But I'll make it more explicit just in case

Reply:
It looks very real 😁